public-html/signup.ctrl.php

```diff
@@ -1,20 +1,22 @@
 <?php
-echo $_POST["formSignUpEmail"];
-echo $_POST["formSignUpPassword"];
-echo $_POST["formSignUpPasswordConf"];
+
+echo $_POST["formSignUpEmail"]; echo '<br>';
+echo $_POST["formSignUpPassword"]; echo '<br>';
+echo $_POST["formSignUpPasswordConf"]; echo '<br>';
 
 $user_email= $_POST["formSignUpEmail"];
-$user_email_pattern ="~^[\w]{1,}[\w.+-]{0,}@[\w-]{2,}([.][a-zA-z]{2,}|[.][\w-]{2,}[.][a-zA-Z]{2,})~";
 $user_password= $_POST["formSignUpPassword"];
-$user_password_pattern ="(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@*$#]).{8,16}";
-$email_validation = preg_match($user_email_pattern,$user_email);
-$password_validation = preg_match($user_password_pattern,$user_password);
+$user_password_confirmation= $_POST["formSignUpPasswordConf"];
 
+$user_email_pattern ='/^[\w]{1,}[\w.+-]{0,}@[\w-]{2,}([.][a-zA-z]{2,}|[.][\w-]{2,}[.][a-zA-Z]{2,})/';
+$user_password_pattern ='/(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@*$#]).{8,16}/';
 
```

Comment:
php  regular expressions need to be wrapped in forward slashed  /my reg exp/ 

example = $my_reg = '/lookforthisstring/'

also use single quotes ' .... only use double quotes when you need to interpolate a value "